src/smt/theory_str_noodler/decision_procedure.cpp

```diff
@@ -896,4 +915,37 @@ namespace smt::noodler {
         return new_eqs;
     }
 
+    /**
+     * @brief Try to replace not contains predicates. In particular, we replace predicates of the form (not_contains lit x) where 
+     * lit is a literal by a regular constraint x notin Alit' where  Alit' was obtained from A(lit) by setting all 
+     * states initial and final. 
+     */
```

Comment:
Should we put these in `.cpp` or `.h` files?

Reply:
I don't know. Personally, I kind of like that in `cpp` files these comments visually splits the functions. I think it improves the orientation in the file.

Reply:
We should probably decide, I usually put them in `.h` files, so now it's quite a mess. @Adda0 do you have any opinion on this?

Reply:
I see that the function declaration in header file has only a minimal documentation comment and the main documentation comment is here. I am strongly against using this pattern. The documentation can go only in header files, otherwise it is useless. IDEs will not show the documentation, it will not get generated, no one will come looking for the documentation to source files. Unless the function is a local-only function in an anonymous namespace, its documentation must be by its declaration, that is, in the header file.

> I kind of like that in cpp files these comments visually splits the functions. I think it improves the orientation in the file.

I understand. However, this is a job of IDEs. Alternatively, you can add some kind of comment separator (`========`, etc.), but again, this should be configured in your IDE of choice, not in the source file.

Reply:
Well, I don't want to fight for the option to keep the comments in `cpp` files, but your arguments are not strong enough:

1. My IDE (vscode) shows the documentation even for function commented in `cpp` file
2. A generation of some documentation from the code is out of the table at least for now. It requires a lot of effort to do it well in order to be truly useful.
3. That might be true for libraries, but still it is a matter of taste. 
4. comment separator is ugly (although it is an option). 

Since we are the only contributors now, the most important reason to consider is convenience of our work. Btw. here is quite nice discussion https://softwareengineering.stackexchange.com/questions/84071/is-it-better-to-document-functions-in-the-header-file-or-the-source-file

Reply:
Ok, since we have different opinions, most likely @jurajsic will decide. Anyway, it is not necessary to decide it now (it should be done in a separate PR).

Reply:
I am for putting them in header files

Reply:
ok, then